speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DatabaseSettingsMenu.cpp

```diff
@@ -0,0 +1,264 @@
+    GfuiVisibilitySet(s_scrHandle, m_privateCertFileDialogControl, m_dbsettings.UseSSL);
+}
+
+/// @brief Saves the settings into the DatabaseSettingsMenu.xml file
+static void SaveSettingsToDisk()
+{
+    // Copies xml to documents folder and then opens file parameter
+    std::string dstStr("config/DatabaseSettingsMenu.xml");
+    char dst[512];
+    sprintf(dst, "%s%s", GfLocalDir(), dstStr.c_str());
+    void* readParam = GfParmReadFile(dst, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Save max time to xml file
+    GfParmSetStr(readParam, PRM_USERNAME, GFMNU_ATTR_TEXT, m_dbsettings.Username);
+    GfParmSetStr(readParam, PRM_PASSWORD, GFMNU_ATTR_TEXT, m_dbsettings.Password);
+    GfParmSetStr(readParam, PRM_ADDRESS, GFMNU_ATTR_TEXT, m_dbsettings.Address);
+    GfParmSetStr(readParam, PRM_PORT, GFMNU_ATTR_TEXT, m_portString);
+    GfParmSetStr(readParam, PRM_SCHEMA, GFMNU_ATTR_TEXT, m_dbsettings.Schema);
+    GfParmSetStr(readParam, PRM_SSL, GFMNU_ATTR_CHECKED, GfuiMenuBoolToStr(m_dbsettings.UseSSL));
+    GfParmSetStr(readParam, PRM_CERT, GFMNU_ATTR_CA_CERT, m_dbsettings.CACertFilePath);
+    GfParmSetStr(readParam, PRM_CERT, GFMNU_ATTR_PUBLIC_CERT, m_dbsettings.PublicCertFilePath);
+    GfParmSetStr(readParam, PRM_CERT, GFMNU_ATTR_PRIVATE_CERT, m_dbsettings.PrivateCertFilePath);
+
+    // Write all the above queued changed to xml file
+    GfParmWriteFile(nullptr, readParam, "DatabaseSettingsMenu");
+}
+
+/// @brief Saves the settings and activates the next menu
+static void SaveSettings(void* /* dummy */)
+{
+    // Save settings in the ResearcherMenu.xml
+    SaveSettingsToDisk();
+    SMediator::GetInstance()->SetDatabaseSettings(m_dbsettings);
+
+    // Go to the next screen
+    GfuiScreenActivate(s_nextHandle);
+}
+
+static void OnActivate(void* /* dummy */)
+{
+    tDbControlSettings control;
+    control.Username = m_usernameControl;
+    control.Password = m_passwordControl;
+    control.Address = m_addressControl;
+    control.Port = m_portControl;
+    control.PortString = m_portString;
+    control.Schema = m_schemaControl;
+    control.UseSSL = m_useSSLControl;
+    control.CACertificateButton = m_caCertFileDialogControl;
+    control.CACertificateLabel = m_caCertDialogLabel;
+    control.PublicCertificateButton = m_publicCertFileDialogControl;
+    control.PublicCertificateLabel = m_publicCertDialogLabel;
+    control.PrivateCertificateButton = m_privateCertFileDialogControl;
+    control.PrivateCertificateLabel = m_publicCertDialogLabel;
+    LoadDBSettings(s_scrHandle, m_dbsettings, control);
+    SynchronizeControls(s_scrHandle, m_dbsettings, control);
+}
+
+/// @brief Returns to the main menu screen
+static void GoBack(void* /* dummy */)
+{
+    GfuiScreenActivate(DataSelectionMenuInit(s_scrHandle));
+}
+
+/// @brief Select a certificate file and save the path
+static void SelectCert(int p_buttonControl, int p_labelControl, char* p_normalText, char* p_filePath, const wchar_t** extensions)
```

Comment:
```suggestion
static void SelectCert(int p_buttonControl, int p_labelControl, char* p_normalText, char* p_filePath, const wchar_t** p_extensions)
```

Also p_normalText should probably be const. (When you call this you pass in const char* which gets converted, which is invalid: `ISO C++11 does not allow conversion from string literal to 'char *'`)

Reply:
made it const and changed to p_extensions